python/ray/data/dataset.py

```diff
@@ -988,7 +988,7 @@ def repartition(
         Examples:
             >>> import ray
             >>> ds = ray.data.range(100)
-            >>> ds.repartition(10).num_blocks()
+            >>> ds.repartition(10)._plan.initial_num_blocks()
```

Comment:
If we don't want to expose `num_blocks` to user, I figure we don't want to expose `Dataset._plan.initial_num_blocks` either?